### PR TITLE
April Fools Day replaces the overflow role with Clowns

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
:cl: Kor
code: Jobs other than Assistant may now be assigned by admins or events as the overflow role
add: April Fools now sets the overflow role as Clown.
/:cl:

Changed the overflow role stuff to use a var instead of always Assistant, meaning admins or coders can pick whatever job they want as the overflow.

April fools now lasts the 1st to the 5th, and sets clown as the overflow role.

### CODE REVIEW

File: code/modules/client/preferences.dm [hunk 1]
```diff
@@ -569,7 +569,7 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 				var/available_in_days = job.available_in_days(user.client)
 				HTML += "<font color=red>[rank]</font></td><td><font color=red> \[IN [(available_in_days)] DAYS\]</font></td></tr>"
 				continue
-			if((job_civilian_low & ASSISTANT) && (rank != "Assistant") && !jobban_isbanned(user, "Assistant"))
+			if((job_civilian_low) && (rank != SSjob.overflow_role) && !jobban_isbanned(user, SSjob.overflow_role))
```
Comment (KorPhaeron) [inline code/modules/client/preferences.dm]: Need to figure out a way to properly check the overflow role flag for the first part of the if

I could set it manually as its own var but that would probably result in admins setting one var and not the other and breaking job setup

Comment (ExcessiveUseOfCobblestone) [reply code/modules/client/preferences.dm]: simple semihacky solution: Create a helper proc that does both at once for admins.

**Comment by AlexTheSysop42** [comment]
>_Nervous laughter_

Heavy stuff on 1st day only (if even), lighter stuff on 2-5

**Comment by Jalleo** [comment]
Cue admins causing all assistants to start as chefs instead and then playing the song there are too many chefs in the kitchen after one of them says that.

**Comment by ExcessiveUseOfCobblestone** [comment]
@Wilchenx not a job.

**Comment by Wilchenx** [comment]
why do you have to be correct

😢

**Comment by KorPhaeron** [comment]
Possible to set it to borg though if an admin is feeling particularly zany

**Comment by ExcessiveUseOfCobblestone** [comment]
borg plus sqdl hmmm

File: code/modules/holiday/holidays.dm [hunk 2]
```diff
@@ -149,10 +149,11 @@
 /datum/holiday/april_fools
 	name = APRIL_FOOLS
 	begin_day = 1
-	end_day = 2
+	end_day = 5
```
Comment (Jordie0608) [inline code/modules/holiday/holidays.dm]: I feel this is a bad idea.

Comment (Iamgoofball) [reply code/modules/holiday/holidays.dm]: April Fools and Easter are sharing a date this year, this is pretty needed.

**Comment by kevinz000** [comment]
image

**Comment by Robustin** [comment]
>Guys I don't want to make April Fools unplayable
>Lets make it even more obnoxious and several times longer

Jesus Kor, in b4 "I forgot they all start with musical instruments so once more than 10 join the station becomes and incoherent cacophony of squeaking midi instruments".

**Comment by ShizCalev** [comment]
# **APRIL FOOLS.**

**Comment by KorPhaeron** [comment]
I forgot they all start with musical instruments

**Comment by KorPhaeron** [comment]
Thank you for reminding me to toggle that off

**Comment by Iamgoofball** [comment]
@KorPhaeron but kor, the clown cannot honk if you disable his hand

**Pull request merged**